web-server/sgx-wallet-impl/src/schema/actions.rs

```diff
@@ -57,6 +61,46 @@ impl From<UnlockWalletError> for OpenWalletResult {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Debug)] // core
+#[derive(Deserialize, Serialize)] // serde
+#[derive(Zeroize, ZeroizeOnDrop)] // zeroize
+pub struct PinReset {
+    pub wallet_id: WalletId,
+    #[zeroize(skip)]
+    // Interior mutability pattern: needed to obtain a draining iterator
+    pub auth_attempts: RefCell<WalletAuthMap>,
```

Comment:
Can you explain why we need to obtain a draining iterator here or use "RefCell"?

Reply:
Does the latest iteration suffice?

Reply:
Yes